src/components/MultiHopTrade/components/TradeInput/components/ManualAddressEntry.tsx

```diff
@@ -39,16 +38,24 @@ export const ManualAddressEntry: FC = memo((): JSX.Element | null => {
   const isYatSupported = isYatFeatureEnabled && isYatSupportedByReceiveChain
 
   const isSnapInstalled = useIsSnapInstalled()
-  const walletSupportsBuyAssetChain = useWalletSupportsChain({
-    chainId: buyAssetChainId,
-    wallet,
-    isSnapInstalled,
-  })
-  const shouldShowManualReceiveAddressInput = !walletSupportsBuyAssetChain
+
+  const useReceiveAddressArgs = useMemo(
+    () => ({
+      fetchUnchainedAddress: Boolean(wallet && isLedger(wallet)),
+    }),
+    [wallet],
+  )
+
+  const isHolisticRecipientAddressEnabled = useFeatureFlag('HolisticRecipientAddress')
+  const activeQuote = useAppSelector(selectActiveQuote)
+  const { manualReceiveAddress } = useReceiveAddress(useReceiveAddressArgs)
+  // If we have a quote, then we were able to get it, meaning we do have a receive address, either wallet default or custom
+  // If the flag is on however, we should hide this as soon as we get a quote, since the new recipient address component will kick in
+  // in <ReceiveSummary />
```

Comment:
This is risky, but is correct as per our flow.

If we want to tighten the checks, logic would look a bit ugly as we would need to not display this once there's an active quote *and* a manual or default wallet address, *or* if there is no *default wallet* receive address

Reply:
Disregard me, this was *not* correct: 2a8f818857af7ccf16177b70d6ec5500f2f6b1eb